docs/specification.md

```diff
@@ -687,12 +708,11 @@ Parameter (in Michelson):
 (bytes %drop_proposal)
 ```
 
-- Delete a finished and accepted proposal that is not flushed. Tokens frozen for this
-proposal are returned to the proposer and voters in full. The decision lambda is skipped.
-- This entrypoint should only be used when there is a proposal that is stuck due to having a
-failing decision lambda.
-- Fails with `NOT_ADMIN` if the sender is not the administrator.
-
+- Delete a proposal when its `proposal_expired_time` is reached.
+- The proposer can call this entrypoint to delete his/her own proposal regardless of `proposal_expired_time`.
+- The `guardian` can call this entrypoint to delete any proposals at anytime regardless of `proposal_expired_time`.
+- Fails with `DROP_PROPOSAL_CONDITION_NOT_MET` when none of the conditions above are met.
+- Tokens that are frozen for this proposal are returned to the proposer and voters as if the proposal is rejected regardless of the actual votes.
```

Comment:
Shouldn't we also say that

>if called on an expired proposal, the paid fee is not returned to the proposer

Reply:
With the current implementation, it treated the return token as if it is a rejected proposal. So it follows the `rejected_proposal_return_value`. My discussion here: https://serokell.slack.com/archives/C019PCGKTK2/p1620821574002100?thread_ts=1620761952.014000&cid=C019PCGKTK2

Now, I'm not sure if I have to state fully about this slash logic in here or not, since it is already stated in the `flush` entrypoints, so I simply state it "...  as if the proposal is rejected". Not sure if this could cause confusion or not.

Reply:
I think the current phrasing is correct, as we now established that dropped proposals are to be treated as failed.

Reply:
I think the current phrasing is correct, as we now established that dropped proposals are to be treated as failed.